_plugins/to_numbering_filter.rb

```diff
@@ -1,15 +1,19 @@
+def clicks_to_numbering(clicks)
```

Comment:
I'd keep this as a class method of `ToNumberingFilter` to avoid polluting the global namespace.

Reply:
Can you tell me how to do that properly? I kept it in there but I couldn't figure out how to access it from outside. I saw some incantation that involved `extend self` that solved my problem, but because I didn't know what that means I was a little scared to do it 😆

Reply:
TL;DR: the trick is remembering that inside any object
* `extend X` mixes in methods of a module X as *class* methods,
* `include X` mixes in methods of a module X as *instance* methods.

A module is itself an object (precisely, an instance of the `Module` class) so you can do that to modules too.

Consider this:

```ruby
module Moo
  extend self
  def foo; end
end
```

You're mixing in **all** the methods of the module `self` (which is `Moo`) into `Moo` as class methods, therefore making `foo` available as `Moo.foo`.

You can also be more specific:

```ruby
module Moo
  def foo; end
  module_function :foo
end
```

Or you can be **way more explicit**, _wink wink_, if you want someone else to understand what the heck is going on:

```ruby
module ToNumberingFilter
  def to_numbering(clicks)
    self.class.clicks_to_numbering(clicks)
  end
  def self.clicks_to_numbering(clicks)
    # ...
  end
end
# => ToNumberingFilter.clicks_to_numbering can now be used everywhere
```

Ruby is less mysterious once you realize they're really not joking about the "everything is an object" motto. Of course that's a loaded gun. :laughing:

Reply:
Thanks, that was a helpful explanation! I've addressed this now in https://github.com/jonsterling/math/commit/463a78671bb07ea89a2735d8d0e8dec1877b193d

Reply:
By the way, the reason why in

```ruby
module Moo
  def met; end
end
```

the method `:met` isn't publicly accessible as `Moo.met` is that it `:met` is an _instance_ method of `Moo`, which is an instance of class `Module` but also a singleton object which cannot really be instanced (that's why it acts basically like a namespace).

In short, it's an instance method of an object you can't instance.

Reply:
Hmm, actually I noticed that the `module_function :foo` has a very different behavior in this case from `extend self`. Not sure exactly what happened, but I will switch to the other.

Reply:
Huh. Weird! There might be some subtle interference w/ Liquid (see "loaded gun" above).
Let's just use `extend` if it works as you expect. :shrug: